File: app/src/main/java/com/galtashma/parsedashboard/ListPreferenceStore.java
package com.galtashma.parsedashboard;

import com.appizona.yehiahd.fastsave.FastSave;

import java.util.ArrayList;
import java.util.List;

public class ListPreferenceStore {
    private List<String> list;
    private String prefId;

    public ListPreferenceStore(String prefId) {
        this.prefId = prefId;
        list = load();
    }

    public List<String> getList() {
        return list;
    }

    public void add(String key) {
        if (!exists(key)) {
            list.add(key);
            save();
        }
    }

    public void remove(String key) {
        if (list.contains(key)) {
            list.remove(key);
        }
        save();
    }

    public void reset() {
        list = new ArrayList<>();
        save();
    }

    public boolean exists(String key) {
        return list.contains(key);
    }

    public boolean isEmpty() {
        return list.isEmpty();
    }

    public int size() {
        return list.size();
    }

    private void save() {
        FastSave.getInstance().saveObjectsList(prefId, list);
    }

    private List<String> load() {
        List<String> l = FastSave.getInstance().getObjectsList(prefId, String.class);
        if (l != null) {
            return l;
        }
        return new ArrayList<>();
    }
}

File: app/src/main/java/com/galtashma/parsedashboard/ListPreferenceStore.kt
package com.galtashma.parsedashboard

import com.appizona.yehiahd.fastsave.FastSave

/**
 * Created by gal on 3/16/18, rewritten by Cyb3rKo on 05/12/22.
 */

class ListPreferenceStore(private val prefId: String) {
    var list = load()

    fun add(key: String) {
        if (!exists(key)) {
            list.add(key)
            save()
        }
    }

    fun remove(key: String) {
        if (list.contains(key)) {
            list.remove(key)
        }
        save()
    }

    fun reset() {
        list = mutableListOf()
        save()
    }

    fun exists(key: String) = list.contains(key)

    fun isEmpty() = list.isEmpty()

    fun size() = list.size

    private fun save() {
        FastSave.getInstance().saveObjectsList(prefId, list)
    }

    private fun load(): MutableList<String> {
        val l = FastSave.getInstance().getObjectsList(prefId, String::class.java)
        if (l != null) {
            return l
        }
        return mutableListOf()
    }
}
